File: client/core/src/com/mygdx/bomberman/entities/MapObject.java
package com.mygdx.bomberman.entities;

public abstract class MapObject {
  private Position position;

  public MapObject() {
    this.position = new Position();
  }

  public MapObject(final Position position) {
    this.position = position;
  }

  public MapObject(final float x, final float y) {
    position = new Position(x, y);
  }

  public Position getPosition() {
    return position;
  }

  public void setPosition(final Position position) {
    this.position = position;
  }
}

File: server/src/main/kotlin/com/bomberman/entity/mapobject/MapObject.kt
package com.bomberman.entity.mapobject

import com.bomberman.entity.Position

abstract class MapObject(
        open var position: Position
) {
    constructor() : this(Position())

    val topLeft: Position
        get() = Position(position.x, position.y + textureHeight - 0.01f)

    val botRight: Position
        get() = Position(position.x + textureWidth - 0.01f, position.y)

    abstract val textureWidth: Int;
    abstract val textureHeight: Int;
}